### PR TITLE
Milestone2 folderstructure

### PR DESCRIPTION
**Pull request by IBUNHABIBU**
- Changes required from code review
   - implement remove book by filtering the selected book id
   - removed the console.log
   - removed the comment

### CODE REVIEW

**Review by Berabjesus — CHANGES_REQUESTED** [review]
# Status: INVALID :red_circle: 
Hey @IBUNHABIBU congrats on getting this far,
**However, I am afraid I can't proceed to check your project because the linter checks are failing**
 

- [ ] Please fix all linter-related problems and ask for a new CRR.

**If you are stuck at any point or have any questions, let me know by leaving a comment here and mentioning me. I will be happy to help. HAPPY CODING :computer:[Bereket Beshane](https://bereketababu.me)** | [LinkedIn](https://linkedin.com/in/bereketbeshane) | [GitHub](https://github.com/Berabjesus)

**Review by mikenath223 — CHANGES_REQUESTED** [review]
### Hello @IBUNHABIBU  👋

**You've done pretty well here, you've set up reducers and actions with initial state. 👌**

However, let's make a couple of changes.

- Please refer to comments on the code.

### **STATUS: CHANGES REQUESTED** 👷‍♂️

Kindly fix these issues and request another review👍
Feel free to contact me on slack(michgolden) if you need further clarification.

**HAPPY CODING 🎉💻🎉**

**Michgolden Ukeje (CR)**

File: src/actions/index.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import { ADD_BOOK, REMOVE_BOOK } from './constants';
+
+export const addBook = (book) => ({
+  type: ADD_BOOK,
+  payload: book,
+});
+
+export const removeBook = () => ({
+  type: REMOVE_BOOK,
+});
```
Comment (mikenath223) [inline src/actions/index.js]: - [ ] Please include the book payload also.

File: src/index.js [hunk 2]
```diff
@@ -1,11 +1,18 @@
 import React from 'react';
 import ReactDOM from 'react-dom';
+import { createStore } from 'redux';
+import { Provider } from 'react-redux';
 import App from './App';
 import reportWebVitals from './reportWebVitals';
+import booksReducer from './reducers/book';
 
+const store = createStore(booksReducer);
+console.log(store.getState());
```
Comment (mikenath223) [inline src/index.js]: - [ ] Kindly remove this console.log. Debugging code shouldn't be pushed to production.

File: src/reducers/book.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+},
+];
+const booksReducer = (state = initialState, { type }) => {
+  switch (type) {
+    case ADD_BOOK:
+      return [...state, {
+        id: 2,
+        title: 'prince',
+        category: 'Movie',
+      }];
```
Comment (mikenath223) [inline src/reducers/book.js]: - [ ] In the ADD_BOOK conditional please return type.payload not a default book: `[...state, type.payload]`

File: src/reducers/book.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    case REMOVE_BOOK:
+      return { };
+    default:
```
Comment (mikenath223) [inline src/reducers/book.js]: - [ ] Also, kindly return a new array of books with the book to be removed filtered out of it. It should return the filtered array, not an object.

File: src/App.js [hunk 5]
```diff
@@ -1,11 +1,14 @@
+import React from 'react';
+// import { useSelector } from 'react-redux';
```
Comment (mikenath223) [inline src/App.js]: - [ ] Kindly remove this commented code line. If you need a previous version of the code you can refer to Git history.

**Review by Forison — APPROVED** [review]
## Status:: Approved ✔️

# hello @IBUNHABIBU 

Good job on addressing the issues raised in the previous review🤗, ie:  
- Removing console.log
- Removing commented code
- Filtering state if the remove book block

Kindly go ahead and merge the branch and proceed to your next task.

#### Happy coding😎

Addo Forison

**Pull request merged**